sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/higherOrderFunctions.scala

```diff
@@ -30,13 +30,14 @@ import org.apache.spark.sql.types.DataType
  * so we need to resolve higher order function when all children are either resolved or a lambda
  * function.
  */
-case class ResolveHigherOrderFunctions(catalog: SessionCatalog) extends Rule[LogicalPlan] {
+case class ResolveHigherOrderFunctions(catalogManager: CatalogManager)
+  extends Rule[LogicalPlan] with LookupCatalog {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveExpressions {
-    case u @ UnresolvedFunction(fn, children, false, filter, ignoreNulls)
+    case u @ UnresolvedFunction(AsFunctionIdentifier(ident), children, false, filter, ignoreNulls)
```

Comment:
If the current catalog is a custom v2 catalog, and the function name has only a single part, seems `AsFunctionIdentifier` always return None, while we should treat it as a temp function first.

Reply:
@cloud-fan Thanks. Yes you are right, it will fail in this case. I think we can perhaps get rid of `AsFunctionIdentifier` and following similar logic as you pointed above:
```scala
case u @ UnresolvedFunction(nameParts, children, false, filter, ignoreNulls)
    if hasLambdaAndResolvedArguments(children) =>
  val CatalogAndIdentifier(catalog, ident) = nameParts
  if (nameParts.length == 1 || CatalogV2Util.isSessionCatalog(catalog)) {
   ...
  }
}
```

Reply:
Neverwind the above. I've updated `AsFunctionIdentifier` to handle the built-in/temp function case so the logic here is much simpler now.